<launch>
	<include file="$(find duckietown)/machines"/>
	<arg name="pkg_name" value="led_detection"/>
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>
	<arg name="local" default="true" doc="true for running on laptop. false for running on vehicle."/>
	<arg name="config" default="baseline" doc="Specify a config."/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>
	<arg name="node_name" default="LED_detector_node_gazebo"/>
	<arg name="continuous" default="True"/> <!-- manual trigger required if false [interactive mode] -->
               <arg name="location_config" default="location_config"/>
	<!-- Run detector on remote (vehicle) -->
	<node unless="$(arg local)" ns="$(arg veh)" machine="$(arg veh)" pkg="$(arg pkg_name)" type="LED_detector_node_gazebo.py" name="LED_detector_node" output="screen" clear_params="true" required="true">
		<rosparam command="load" file="$(find duckietown)/config/$(arg config)/led_interpreter/LED_protocol.yaml"/>
		<rosparam command="load" file="$(find duckietown)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
		<param name="continuous" type="bool" value="$(arg continuous)" />
	</node>	
 <!-- Local -->
<node if="$(arg local)"  pkg="$(arg pkg_name)" type="LED_detector_node_gazebo.py" name="LED_detector_node_$(arg veh)" output="screen" clear_params="true" required="true">
		<remap from="~tag"  to="$(arg veh)/apriltags_postprocessing_node/apriltags_out"/>
                              <remap from="traffic_light/raw_led_trafficlight" to="traffic_light_node_gazebo/raw_led_trafficlight"/>
                              <remap from="~raw_led_detection" to="$(arg veh)/LED_detector_node_gazebo/raw_led_detection" />
                              <rosparam command="load" file="$(find duckietown)/config/$(arg config)/led_interpreter/LED_protocol.yaml"/>
		<rosparam command="load" file="$(find duckietown)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
		<param name="continuous" type="bool" value="$(arg continuous)" /> 
                               <param name="veh" type="string" value="$(arg veh)"/>                  
	                <rosparam command="load" file="$(find duckietown)/config/$(arg config)/led_interpreter/$(arg location_config).yaml"/>


</node>
</launch>
